web30/src/erc20_utils.rs

```diff
@@ -42,35 +37,43 @@ impl Web3 {
             }
         });
 
+        // Check if the allowance remaining is greater than half of a Uint256- it's as good
+        // a test as any.
+        Ok(allowance)
+    }
+
+    /// Checks if `spender` is approved to spend a large amount of `erc20` tokens held by `owner`
+    /// Allowances are commonly used by protocols to manage tokens on behalf of users,
+    /// users simply approve a contract and then call the contract to perform actions
+    pub async fn check_erc20_approved(
```

Comment:
why not have the user specify the expected allowance? Because huge allowances are generally bad policy. But it's what we've expected in this library up until now.

Reply:
Marked deprecated